### ISSUE TITLE
Rollback issue about mending experience level, 

### ISSUE THREAD

**Issue by ghost** [pl3xgaming/Purpur#683]
### Expected behavior

The experience decreases sequentially to the anvil and mending.

### Observed/Actual behavior

Mending is operated as an experience before using the anvil.
30lv(1,395xp) -> use anvil and level decrease to 26lv -> use mending with shift+right click -> roll back to 30lv and use exp for mending (in my case, its 5xp)

### Steps/models to reproduce

1. Enable shift-right-click-repairs-mending-points: 5 
2. Use an anvil to consume experience.
3. Repair(with mending) the equipment with Shift + Right Click like option 1.
4. In the experience value before using the anvil, the experience value decreases as much as used for bending.

### Purpur version

This server is running Purpur version git-Purpur-1394 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: acb8a26 on ver/1.17.1)
You are running the latest version
Previous version: git-Purpur-1393 (MC: 1.17.1)

### Agreements

- [X] I am running the latest version of Purpur available from https://purpur.pl3x.net/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Purpur.

**Comment by BillyGalbreath** [comment]
I don't understand what it is you're trying to say. Elaborate on this quite a bit more. [ELI5](https://www.dictionary.com/e/slang/eli5/)

**Comment by encode42** [comment]
Been a month since a response
